src/dataframe/dataframe.jl

```diff
@@ -822,7 +823,34 @@ end
 
 Base.deleteat!(df::DataFrame, inds::Not) = deleteat!(df, axes(df, 1)[inds])
 
-function _deleteat!_helper(df::DataFrame, drop)
+function _isincreasing(v::AbstractVector)
+    length(v) < 2 && return true
+    last = v[begin]
+    for i in firstindex(v)+1:lastindex(v)
+        cur = @inbounds x[i]
+        cur > last || return false
+        last = cur
+    end
+    return true
+end
+
+function _uniquesorted(v::AbstractVector)
+    x = sort(v)
+    # here we know that length of v is at least 2
+    last = v[begin]
+    u = eltype(v)[last]
+    for i in firstindex(v)+1:lastindex(v)
+        cur = @inbounds x[i]
+        cur == last && continue
+        push!(u, cur)
```

Comment:
`push!` is slow, it would be faster to assign and `resize!` at the end, like `unique!` does (https://github.com/JuliaLang/julia/blob/9a2f5ae0531524e60b3308d7695100b14801534e/base/set.jl#L315). Probably calling `unique!(sort(x))` would be enough. But if that's the case, given that it's easy for users to do that manually, I wonder whether it's really necessary to do it automatically..